src/opm/parser/eclipse/EclipseState/Schedule/Well/WellTestState.cpp

```diff
@@ -130,7 +130,11 @@ namespace Opm {
                     if (well_config.num_test == 0 || (well.num_attempt < well_config.num_test)) {
                         well.last_test = sim_time;
                         well.num_attempt ++;
-                        output.emplace_back(std::make_pair(well.name, well.reason));
+                        if (std::find(output.begin(), output.end(), well.name) == output.end()) {
+                          // create a unique list
+                          output.emplace_back(well.name);
+                        }
```

Comment:
This is quadratic in the number of wells.  Is that a reasonable cost here?  If not, I'd probably just push back the `well.name` and use `std::sort()` and `std::unique()` at the end in order to guarantee uniqueness.